From db94baa4bf88f0e21fe97d1c7b766200bc267381 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 13 Oct 2014 15:48:07 +0800
Subject: [PATCH] MLK-9684-8: ARM: dts: imxqdl-sabreauto: refine fm device
 tree

As the radio machine drive use the codec_of_node, so add si476x-codec
node for this usage.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
(cherry picked from commit 89a4342b5d6ee21173566fab6ae017c660a23620)
---
 arch/arm/boot/dts/imx6qdl-sabreauto.dtsi | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
index 9402bd5..4ead066 100644
--- a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
@@ -236,7 +236,7 @@
 			   "fsl,imx-tuner-si476x";
 		model = "imx-radio-si4763";
 		ssi-controller = <&ssi2>;
-		fm-controller = <&si4763>;
+		fm-controller = <&si476x_codec>;
 		mux-int-port = <2>;
 		mux-ext-port = <5>;
 	};
@@ -447,6 +447,10 @@
 		vd-supply = <&reg_si4763_vd>;
 		vio1-supply = <&reg_si4763_vio1>;
 		vio2-supply = <&reg_si4763_vio2>;
+
+		si476x_codec: si476x-codec {
+			compatible = "si476x-codec";
+		};
 	};
 };
 
-- 
1.8.0

